hal/tests/UNITTESTS/doubles/CMakeLists.txt

```diff
@@ -24,6 +32,8 @@ target_link_options(mbed-stubs-hal
 
```

Comment:
Given that `--coverage` is globally set in UNITTESTS/CMakeLists.txt, can we remove it from here (the four lines above this)? I also asked in #14773 about the platform stubs - we can investigate and align them.

Reply:
Initially, we added target_link_libraries `mbed-stubs-hal` with `gcov` to resolve the dependency issue related to coverage, but this didn't work with Armclang (build failure) so Rob proposed to use the target link option with --coverage to resolve that issue refer commit.https://github.com/ARMmbed/mbed-os/pull/14426/commits/10625b35b1818520deb5cd51369265fcffcc4022

Reply:
I have removed --coverage link options changes and tested with ubuntu Linux , please re-review